### ISSUE TITLE
more events with same id, cannot [remove_event]

### ISSUE THREAD

**Issue by hladkyjakub** [wesnoth/wesnoth#9342]
### Game and System Information

- **Version:** 1.18

### Description of the bug

I cannot [remove_event] my event (for some reason).
I can create many events with same id (no idea how is that even possible).
Everything works once game is saved and loaded, until the [event] is created again.

(event run multiple times on triggered)
From game inspector:
![Screenshot_20240912_212421](https://github.com/user-attachments/assets/b9128b71-2d32-4cf7-bc4a-52c529a9a80f)

### Steps to reproduce the behavior

I have no clue what particular part of my code causes the bug, and I cannot reproduce it other than with a save file.

### Expected behavior

Being able to remove the event, no duplicate id's.

**Comment by gfgtdf** [comment]
Might be the same cause as https://github.com/wesnoth/wesnoth/issues/9225

**Comment by soliton-** [comment]
With no way to reproduce this it is unlikely that anyone can investigate this.

**Comment by gfgtdf** [comment]
Assuming this is the same cause as #9225 this should be fixed in 1.18, would be nice If you could test it since we cannot check your specific case

**Comment by hladkyjakub** [comment]
> Assuming this is the same cause as #9225 this should be fixed in 1.18, would be nice If you could test it since we cannot check your specific case

I've tested it on 1.18.0 ... what 1.18 should it be fixed for?

**Comment by gfgtdf** [comment]
1.18 latest git branch, not released yet. (I'm actually not sure whether you can build Wesnoth I keep forgetting who can and who can't. But 1.19.4 has the same commit so testing this on 1.19.4 would also help a lot here

**Comment by gfgtdf** [comment]
Also: does anyone now whether the GitHub ci produces artifacts?iirc the old Travis used to, and it was kinda helpful when we asked people to test stuff.

**Comment by hladkyjakub** [comment]
> 1.18 latest git branch, not released yet. (I'm actually not sure whether you can build Wesnoth I keep forgetting who can and who can't. But 1.19.4 has the same commit so testing this on 1.19.4 would also help a lot here

I can test on latest 1.18 IMHO (how would I work on cpp if I couln't build Wesnoth?)

...

What is github ci?

**Comment by soliton-** [comment]
Yes, you get a flatpak bundle for linux, a disk image for macos and an installer for windows.

**Comment by hladkyjakub** [comment]
> Yes, you get a flatpak bundle for linux, a disk image for macos and an installer for windows.

What?

**Comment by gfgtdf** [comment]
> I can test on latest 1.18 IMHO (how would I work on cpp if I couln't build Wesnoth?)

Yes my mad, i always mistake who did what here. If you could test on 1.18 latest commit that would be nice.

> What is github ci?

When commiting (and an making prs) The wesnoth source code is checked, built and tested (the green check next to the commits) . A service that does this is called a CI (Continuous Integration). But our CI doesn't only run the test, it also uploads the binaries it creates. For example for the the commit https://github.com/wesnoth/wesnoth/commit/62685dc523089eec9d45625f6470f6bb12dc43d0 , the CI creates also a windows installer here https://github.com/wesnoth/wesnoth/actions/runs/10913089418/artifacts/1945490569 . This is quite useful if we ask someone who doesn't know how to build wesnoth to test at a specific commit since we can then easily give them an installer for thecorresponding version.

(Now i know this is not that relevent or you since you can build if by yourself, i stil lwanted to explain since you asked.)

> Yes, you get a flatpak bundle for linux, a disk image for macos and an installer for windows.

Oh ok, i was looking at the windows build not the mingw built that's why i didnt find it at first.

**Comment by hladkyjakub** [comment]
*Noting, that I don't use priority attribute i the event at all. (the default priorty=0 pops in in save file tho)

**Comment by hladkyjakub** [comment]
And for some reason it seems to cap at 7 events with same id.

**Comment by gfgtdf** [comment]
>*Noting, that I don't use priority attribute i the event at all. 

This doesn't really matter, there are events in mainline that use prioity and are always present in any scenario and could cause this bug.

**Comment by sevu** [comment]
Haven't tested latest 1.18 git, but can confirm the bug for 1.18.2 and that it does not occour in latest git master.

**Comment by hladkyjakub** [comment]
Thanks for testing ^^

1.18.3 is tagging in less than 20 days, I'll test there, if the bug perishes, I'll close this issue :D .

**Comment by Wedge009** [comment]
1.18.4 was released very recently, given the age of this as well as sevu's confirmation, I'm guessing this can be closed.
